CalcualtorLogic.js

```diff
@@ -18,16 +18,155 @@ class CalculatorLogic {
+        periodButton.classList.add("button");
+        const periodButtonText = document.createTextNode(".");
+        periodButton.appendChild(periodButtonText);
+        const negativeValueButton = document.createElement("button");
+        negativeValueButton.classList.add("button");
+        const negativeValueButtonText = document.createTextNode("±");
+        negativeValueButton.appendChild(negativeValueButtonText);
+        div4.appendChild(button0);
+        div4.appendChild(periodButton);
+        div4.appendChild(negativeValueButton);
+        wrapperDiv.appendChild(div4);
+        const div5 = document.createElement("div");
+        const multiplyingButton = document.createElement("button");
+        multiplyingButton.dataset.operator = "*";
+        multiplyingButton.classList.add("button");
+        const multiplyingButtonText = document.createTextNode("*");
+        multiplyingButton.appendChild(multiplyingButtonText)
+        const addingButton = document.createElement("button");
+        addingButton.dataset.operator = "+";
+        addingButton.classList.add("button");
+        const addingButtonText = document.createTextNode("+");
+        addingButton.appendChild(addingButtonText)
+        const subtractingButton = document.createElement("button");
+        subtractingButton.dataset.operator = "-";
+        subtractingButton.classList.add("button");
+        const subtractingButtonText = document.createTextNode("-");
+        subtractingButton.appendChild(subtractingButtonText);
+        div5.appendChild(multiplyingButton);
+        div5.appendChild(addingButton);
+        div5.appendChild(subtractingButton);
+        wrapperDiv.appendChild(div5);
+        const div6 = document.createElement("div");
+        const dividingButton = document.createElement("button");
+        dividingButton.dataset.operator = "/";
+        dividingButton.classList.add("button");
+        const dividingButtonText = document.createTextNode("/");
+        dividingButton.appendChild(dividingButtonText);
+        const equalButton = document.createElement("button");
+        equalButton.classList.add("button");
+        const equalButtonText = document.createTextNode("=");
+        equalButton.appendChild(equalButtonText);
+        const clearingButton = document.createElement("button");
+        clearingButton.classList.add("button");
+        const clearingButtonText = document.createTextNode("Clear");
+        clearingButton.appendChild(clearingButtonText)
+        div6.appendChild(dividingButton);
+        div6.appendChild(equalButton);
+        div6.appendChild(clearingButton);
+        wrapperDiv.appendChild(div6);
+        this.hostElem.appendChild(wrapperDiv);
+    };
+    getButtonsRow(buttons) {
+        const div = document.createElement("div");
+        for (let button of buttons) {
+            button = document.createElement("button");
+            button.dataset.number;
+            const buttonText = document.createTextNode();
```

Comment:
you should pass a value to the ` document.createTextNode();`
the value is coming for the array you used when you called the `getButtonsRow` method